### PR TITLE
Issue#96

### PR DESCRIPTION
**Pull request by DhritiShikhar**
Creates a requirements.txt

### CODE REVIEW

**Comment by laxathom** [comment]
Sound good.Will merge it manually to only take the latest.

**Comment by laxathom** [comment]
@ralphbean you mean at line 238?

**Comment by ralphbean** [comment]
> @ralphbean you mean at line 238?

Yes :)

**Comment by DhritiShikhar** [comment]
@ralphbean could you please take a look at the patch?

**Comment by laxathom** [comment]
sound good for real now :)
:+1:

**Comment by ralphbean** [comment]
This still looks incorrect to me.  Once again, the **value** of self.valid_ssh_key is not being interpolated into the regex pattern.

This would only match ssh keys that have the literal string ``"valid_ssh_key"`` in them, which is not what we want.  We want to match keys that have the literal strings ``"rsa"`` or ``"ssh-rsa"`` or ``"ecdsa"``, etc..  So, those strings are in a **variable** named ``valid_ssh_key`` which needs to be [interpolated](https://docs.python.org/2/library/stdtypes.html#string-formatting-operations) into the regex pattern before it is applied.

**Comment by laxathom** [comment]
hm, that's actually what the the .replace() does by adding the "|" which stand for OR.
I did try it myself from a python console and it does works as expected.

**Comment by pypingou** [comment]
@laxathom it's the ``re.match()`` line that's problematic

**Comment by laxathom** [comment]
Ah, stupid me!
Saw the passed string instead of the variable.

File: fas/validators.py [hunk 1]
```diff
@@ -233,7 +235,7 @@ def validate_python(self, value, state):
             if not keyline:
                 continue
             keyline = keyline.strip()
-            validline = re.match('^(rsa|ssh-rsa) [ \t]*[^ \t]+.*$', keyline)
+            validline = re.match('^(valid_ssh_key) [ \t]*[^ \t]+.*$', keyline)
```
Comment (laxathom) [inline fas/validators.py]: you can do something like this: re.match('^({}) [ \t]*[^ \t]+.*$'.format(valid_ssh_key), keyline)

File: fas/validators.py [hunk 2]
```diff
@@ -233,7 +235,7 @@ def validate_python(self, value, state):
             if not keyline:
                 continue
             keyline = keyline.strip()
-            validline = re.match('^(rsa|ssh-rsa) [ \t]*[^ \t]+.*$', keyline)
+            validline = re.match('^({}) [ \t][^ \t]+.$'.format(valid_ssh_key), keyline)
```
Comment (laxathom) [inline fas/validators.py]: you have two "*" missing there. re.match('^({}) [ \t]*[^ \t]+.*$'.format(valid_ssh_key)
My previous comment was an example of a way to interpolate your value ;)

**Comment by laxathom** [comment]
hm, do not copy/paste my line. some caracters got stripped by markdown

**Comment by DhritiShikhar** [comment]
@laxathom could you please check if the patch is alright?

**Comment by pypingou** [comment]
@laxathom ping ?

**Comment by laxathom** [comment]
Sorry for the delay.
Just did a manual merge by squashing your commits.

**Pull request closed without merging**